fee-handler-router/src/lib.rs

```diff
@@ -0,0 +1,240 @@
+// The Licensed Work is (c) 2022 Sygma
+// SPDX-License-Identifier: LGPL-3.0-only
+
+#![cfg_attr(not(feature = "std"), no_std)]
+
+pub use self::pallet::*;
+
+#[cfg(test)]
+mod mock;
+
+#[allow(unused_variables)]
+#[allow(clippy::large_enum_variant)]
+#[frame_support::pallet]
+pub mod pallet {
+	use frame_support::{pallet_prelude::*, traits::StorageVersion};
+	use frame_system::pallet_prelude::*;
+	use sygma_traits::{DomainID, FeeHandler};
+	use xcm::latest::AssetId;
+
+	#[derive(PartialEq, Eq, Clone, Encode, Decode, TypeInfo, RuntimeDebug)]
+	pub enum FeeHandlerType {
+		BasicFeeHandler,
+		DynamicFeeHandler,
+	}
+
+	const STORAGE_VERSION: StorageVersion = StorageVersion::new(0);
+
+	#[pallet::pallet]
+	#[pallet::generate_store(pub (super) trait Store)]
+	#[pallet::storage_version(STORAGE_VERSION)]
+	#[pallet::without_storage_info]
+	pub struct Pallet<T>(_);
+
+	#[pallet::config]
+	pub trait Config: frame_system::Config + sygma_basic_feehandler::Config {
+		type RuntimeEvent: From<Event<Self>> + IsType<<Self as frame_system::Config>::RuntimeEvent>;
+
+		/// Origin used to administer the pallet
+		type BridgeCommitteeOrigin: EnsureOrigin<Self::RuntimeOrigin>;
+
+		/// Fee handlers
+		type BasicFeeHandler: FeeHandler;
+		type DynamicFeeHandler: FeeHandler;
+
+		/// Current pallet index defined in runtime
+		type PalletIndex: Get<u8>;
+	}
+
+	#[pallet::event]
+	#[pallet::generate_deposit(pub (super) fn deposit_event)]
+	pub enum Event<T: Config> {
+		/// When fee handler was set for a specific (domain, asset) pair
+		/// args: [dest_domain_id, asset_id, handler_type]
+		FeeHandlerSet { domain: DomainID, asset: AssetId, handler_type: FeeHandlerType },
+	}
+
+	#[pallet::error]
+	pub enum Error<T> {
+		/// Account has not gained access permission
+		AccessDenied,
+		/// Function unimplemented
+		Unimplemented,
+	}
+
+	/// Mark whether a deposit nonce was used. Used to mark execution status of a proposal.
+	#[pallet::storage]
+	#[pallet::getter(fn handler_type)]
+	pub type HandlerType<T: Config> =
+		StorageMap<_, Twox64Concat, (DomainID, AssetId), FeeHandlerType>;
```

Comment:
I assume that using the combination of `DomainID` and `AssetID` here is because in solidity we identify a fee handler by `_domainResourceIDToFeeHandlerAddress[destinationDomainID][resourceID];`. If so, it makes sense, however, as we discussed, each pallet runtime instance will only bind with one specific domainID, which means `DomainID` is always the same, if so, then why don't we just use `assetID`?

Reply:
Not "each pallet runtime" but "each pallet instance", we can have multiple bridge instances in runtime. But as we discussed, we would refactor it.

Regarding the binding of `DomainID` when setting fees, in my understanding, different fee handlers represent different fee strategies, for the basic fee handler, we set a constant amount, and each dest chain should have a different amount. The fee for Ethereum should be less than the fee for Moonbeam. Correct me if I am wrong.

Reply:
sorry, yeah we should be careful when using those terms 😄, I didn't consider the multiple bridge support refactoring when review this PR, but with this refactoring, yes everything makes sense now.